app/models/conference.rb

```diff
@@ -86,11 +86,11 @@ def user_registered? user
   # After the conference has been successfully updated
   def delete_event_schedules
     if self.start_hour_changed? || self.end_hour_changed?
-      event_schedules = EventSchedule.select do |e|
-        e.event.program == program &&
-        (e.start_time.strftime('%H').to_i < start_hour ||
-        e.end_time.strftime('%H').to_i > end_hour ||
-        (e.end_time.strftime('%H').to_i == end_hour && e.end_time.strftime('%M').to_i > 0))
+      event_schedules = program.event_schedules.select do |event_schedule|
+        event_schedule.event.program == program &&
+        (event_schedule.start_time.strftime('%H').to_i < start_hour ||
```

Comment:
you can use here the `hour` method: :wink: 

`event_schedule.start_time.hour < start_hour`